scripts/main.py

```diff
@@ -1,52 +1,151 @@
+        raise Exception("Tag not found in repostitory")
+    commit = tags[0]
+    version = get_versions_from_tag(tag)
+    if not version:
+        raise Exception("Unable to get version from tag")
+    major = get_major(version)
+    majors_existing = get_newest_version_for_major(get_existing_versions(output_path))
+    if major in majors_existing and version <= majors_existing[major]:
+        return
+    folder_path = output_path / version_to_folder_name(version)
+    if not folder_path.exists():
+        os.makedirs(folder_path.resolve())
+    object_path = folder_path / object_file_name
+    calc_path = folder_path / calc_file_name
+    meta_path = folder_path / meta_info_name
+    updateObjects(commit, object_path)
+    updateCalcs(commit, calc_path)
+    update_meta_info(commit, version, meta_path)
+
+
+def update_versioned_documentation():
+    tags = getTags()
+    tag_versions = {}
+    for tag in tags:
+        version = get_versions_from_tag(tag["name"])
+        if version and version >= first_valid_version:
+            tag_versions[version] = tag["commit"]["sha"]
+    majors_tags = get_newest_version_for_major(list(tag_versions.keys()))
+    majors_existing = get_newest_version_for_major(get_existing_versions(output_path))
+    for major, ver in majors_tags.items():
+        if major in majors_existing and ver <= majors_existing[major]:
+            continue
+        commit = tag_versions[ver]
+        folder_path = output_path / version_to_folder_name(ver)
+        if not folder_path.exists():
+            os.makedirs(folder_path.resolve())
+        object_path = folder_path / object_file_name
+        calc_path = folder_path / calc_file_name
+        meta_path = folder_path / meta_info_name
+        updateObjects(commit, object_path)
+        updateCalcs(commit, calc_path)
+        update_meta_info(commit, ver, meta_path)
+
+
+def update_latest_documentation():
+    folder_path = output_path / "latest"
+    if not folder_path.exists():
+        os.makedirs(folder_path.resolve())
+    object_path = folder_path / object_file_name
+    calc_path = folder_path / calc_file_name
+    meta_path = folder_path / meta_info_name
+    commit = getCommitId("main")
+    updateObjects(commit, object_path)
+    updateCalcs(commit, calc_path)
+    update_meta_info(commit, "latest", meta_path)
+
 
-BRANCH = "main"
-OBJECT_OUTPUT_PATH = "./public/objectsDoc.yaml"
-CALCS_OUTPUT_PATH = "./public/calcs.yaml"
+def update_version_options():
+    package_path = Path("package.json")
+    with open(package_path.resolve(), "r") as f:
+        package = json.load(f)
+    package["contributes"]["configuration"]["properties"][
+        "septic.documentation.version"
+    ]["enum"] = list(map(folder_name_to_option, get_versions(output_path)))
+    package_path = Path("package.json")
+    with open(package_path.resolve(), "w") as f:
+        json.dump(package, f, indent=2)
 
 
-def updateObjects():
+def updateObjects(ref: str, output_path: Path):
```

Comment:
Is `ref` a commit hash or a github ref? It looks like it is the former, in which case `commit` may be a better variable name for readability? Same for `updateCalcs()`

Reply:
It is a github ref. I think both commit and branch is used in the scripts